ludwig/datasets/sst2/__init__.py

```diff
@@ -50,91 +47,10 @@ class SST2(ZipDownloadMixin, MultifileJoinProcessMixin, CSVLoadMixin,
-            Lines = f.readlines()
-            for line in Lines:
-                if line:
-                    split_line = line.split('|')
-                    phrase2id[split_line[0]] = int(split_line[1])
-
-        id2sent = {}
-        with open(os.path.join(self.raw_dataset_path,
-                               'SST-2/original/sentiment_labels.txt')) as f:
-            Lines = f.readlines()
-            for line in Lines:
-                if line:
-                    split_line = line.split('|')
-                    try:
-                        id2sent[int(split_line[0])] = float(split_line[1])
-                    except ValueError:
-                        pass
-
-        def format_sentence(sent):
-            formatted_sent = ' '.join(
-                [w.encode('latin1').decode('utf-8')
-                 for w in sent.strip().split(' ')])
-            formatted_sent = formatted_sent.replace('-LRB-', '(')
-            formatted_sent = formatted_sent.replace('-RRB-', ')')
-            return formatted_sent
-
-        def get_sentence_idxs(split):
-            return set(
-                datasplit_df[
-                    datasplit_df['splitset_label'] == split
-                    ]['sentence_index']
-            )
-
-        def get_sentences(sentences_idxs):
-            criterion = sentences_df['sentence_index'].map(
-                lambda x: x in sentences_idxs
-            )
-            return sentences_df[criterion]['sentence'].tolist()
-
-        def get_sentiment_label(phrase_id):
-            sentiment = id2sent[phrase_id]
-            if sentiment <= 0.4:  # negative
-                return 0
-            elif sentiment > 0.6:  # positive
-                return 1
-            return -1  # neutral
-
-        sentences_df['sentence'] = sentences_df['sentence'].apply(
-            format_sentence
-        )
-
-        splits = {
-            'train': 1,
-            'dev': 3,
-            'test': 2
-        }
-
-        for split_name, split_id in splits.items():
-            sent_idxs = get_sentence_idxs(split_id)
-            sents = get_sentences(sent_idxs)
-            phrase_ids = [phrase2id[phrase] for phrase in sents]
-
-            pairs = []
-            for sent, phrase_id in zip(sents, phrase_ids):
-                sent_label = get_sentiment_label(phrase_id)
-                if sent_label != -1:  # only include non-neutral samples
-                    pairs.append([sent, sent_label])
-
-            final_csv = pd.DataFrame(pairs)
-            final_csv.columns = ['sentence', 'label']
-            final_csv.to_csv(os.path.join(self.raw_dataset_path,
-                                          f'{split_name}.csv'),
-                             index=False)
-
-        super(SST2, self).process_downloaded_dataset()
+    def get_sentiment_label(self, id2sent, phrase_id):
+        sentiment = id2sent[phrase_id]
+        if sentiment <= 0.4:  # negative
```

Comment:
we should put 0.4 and 0.4 in a constants file or make them configs

Reply:
they are a fixed convention of the construction of this specific variant of the SST dataset. We could make them constants, but they won't change so probably there's no need to make them configurable.